mmtrack/evaluation/metrics/mot_challenge_metrics.py

```diff
@@ -198,6 +209,12 @@ def _save_one_video_gts_preds(self, seq: str) -> None:
         pred_file = osp.join(self.pred_dir, seq + '.txt')
 
         pred_tracks = np.array(info['pred_tracks'])
+
+        if self.aflink_cfg is not None:
+            # associate tracks
+            pred_tracks = appearance_free_link(pred_tracks,
+                                               **self.aflink_cfg)
+
         if self.interpolate_tracks_cfg is not None:
             # interpolate tracks
             pred_tracks = interpolate_tracks(pred_tracks,
```

Comment:
Firstly, we may use `postprocess_tracklet_cfg`, which is a list[dict], to denote all postprocessing method for tracklets.

Secondly, we can convert all `interpolate_tracks`, `appearance_free_link`, `gaussian_smoothed_interpolation` from function to class, in order to use `registry`

Here is an example:
```python
# in the __init__ function of metric
self.postprocess_tracklet_cfg = postprocess_tracklet_cfg.copy()
self.postprocess_tracklet_methods = [TASK_UTILS.build(cfg) for cfg in self.postprocess_tracklet_cfg]

# in the _save_one_video_gts_preds function of metric
for postprocess_tracklet_methods  in self.postprocess_tracklet_methods:
    pred_tracks = postprocess_tracklet_methods.forward(pred_tracks)

# for use the registry 
@TASK_UTILS.register()
Class LinearInterpolateTracklets:
   def __init__(xxx, yyy):
        self.xxx=xxx
        self.yyy=yyy
   def forward(pred_tracks):
        ....
       return interpolated_tracks
```
Overall, the postprocessing shares a similar design idea with data pipeline. You can also create a `Compose` class to compose all postprocessing methods, just like the `Compose` in data pipeline.